tests/subsys/spm/secure_services/src/main.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <ztest.h>
+#include <secure_services.h>
+#include <pm_config.h>
+
+static const struct fw_info ns_fw_info = { .magic = { FIRMWARE_INFO_MAGIC } };
+static uint8_t non_secure_buf[128];
+
+static bool is_secure(intptr_t ptr)
+{
+	/* We can not use 'arm_cmse_addr_is_secure here since this firmware
```

Comment:
That's true, although you could probably use the arm_cmse_idau_region_get(ptr)
This should return invalid if accessed from Non-Secure, while, all ram and flash are always covered by one SPU region.
You can check and see if it works. It will make this code more portable.

Reply:
I don't think that will work since that function is only available for code compiled as secure, which this is not.

```
#if defined(CONFIG_ARM_SECURE_FIRMWARE)

/**
  * ...
  */
int arm_cmse_mpu_nonsecure_region_get(uint32_t addr);


```

Reply:
ups, yes, sorry my bad.